File: cli/shared-lib/pkg/models/CommandUsage.go
package models

type CommandUsage struct {
	ID                  string `json:"id"`
	CommandId           string `json:"commandId"`
	CreatorUid          int    `json:"createdByUid"`
	Description         string `json:"description"`
	ExecutablePath      string `json:"executablePath"`
	Arguments           string `json:"arguments"`
	OutputCapturingMode string `json:"outputCapturingMode"`
	OutputFilename      string `json:"outputFilename"`
	OutputParser        string `json:"outputParser"`
}

File: apps/api/src/Domain/Entities/CommandUsage.cs
namespace api_v2.Domain.Entities;

public class CommandUsage : TimestampedEntity
{
    public string Id { get; set; } = string.Empty;

    public string CommandId { get; set; } = string.Empty;

    public int CreatedByUid { get; set; }

    public User? CreatedBy { get; set; }

    public string? Description { get; set; }

    public string? ExecutablePath { get; set; }

    public string? DockerImage { get; set; }

    public string? Arguments { get; set; }

    public string? OutputCapturingMode { get; set; }

    public string? OutputFilename { get; set; }

    public string? OutputParser { get; set; }
}
